Modix.Bot/Modules/AnnounceModule.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Threading.Tasks;
+using Discord;
+using Discord.Commands;
+using Modix.Services.CommandHelp;
+using Modix.Services.Mentions;
+
+namespace Modix.Bot.Modules
+{
+    [Name("announce")]
+    [Summary("Makes an announcement")]
+    [HelpTags("announce", "$")]
+    public class AnnounceModule : ModuleBase
+    {
+        private IMentionService _mentionService { get; }
+
+        public AnnounceModule(IMentionService mentionService)
+        {
+            _mentionService = mentionService;
+        }
+
+        [Command]
+        [Alias("$")]
+        [Summary("Makes an announcement in the currnet channel")]
+        public async Task MakeAnnouncementAsync([Summary("The role to mention")] IRole role,
+            [Remainder, Summary("The message to send as part of the announcement.")]
+            string message)
+        {
+            // Send message
+            await _mentionService.MentionRoleAsync(role, Context.Channel, message);
+
+            // Clean up
+            await Context.Message.DeleteAsync();
```

Comment:
I'm wondering if it might be nice to only delete the message if the service was successfully able to create the announcement (e.g. it didn't fail because the supplied role wasn't marked as mentionable).

Reply:
Should I change `MentionRoleAsync` to return `Task<bool>` or is there a preferred way to do this?

Reply:
That would probably work. I can't think of any more complex information that we'd want to return.

Reply:
That's done. Let me know if something returns something unexpected.